### PR TITLE
Ravestate v0.6.0

### PR DESCRIPTION
**Pull request by josephbirkner**
- [x] 💡PersonalQA [#79]
- [x] ⚛️ Symbols for Property/Signal names [#40]
- [x] 📡 Activation visualisation in UI with SocketIO
- [x] 📦 Dockerfile w/ documentation
- [x] 🍺 Fillers are now persistently exciting [#85]
- [x] 👂 RoboyIO w/ tts/stt-exclusive listen loop
- [x] 📈 ~~77%~~ 76% test coverage [#56, ...]
- [x] ✅ Create minimal RoboyQA nodes automatically [#39] [#88]
- [x] 🚫 Fix for conjunctions acquiring from different causal groups [#57] 
- [x] 👻 Fixes for zombie spikes, death_clock race conditions due to stuck conjunct causal chains [#52] 
- [x] 🌵 Fix for uppercase question parsing [#81] 
- [x] 🏭 Fix spaces for wildtalk outputs. [#80]
- [x] 📝 Fix terminal color literals in reggol logs [#68]
- [x] 👳‍♀️ Akinator polished [#72] - but now broken due to API changes 😞 
- [x] ↕️ Compatibility w/ spacy 2.1.0! Make sure to `pip uninstall en_core_web_sm`, so that spacy can install it's 2.1 model package.

### CODE REVIEW

File: modules/ravestate_phrases_basic_en/__init__.py [hunk 1]
```diff
@@ -3,3 +3,19 @@
...
+intent_facts = "facts"
+intent_farewells = "farewells"
+intent_fillers = "fillers"
+intent_flattery = "flattery"
+intent_greeting = "greeting"
+intent_jokes = "jokes"
+intent_offer_fact = "offer-fact"
+intent_prompt_qa = "question-answering-prompt"
+intent_topic_change = "topicchange"
+intent_unclear = "unclear"
```
Comment (Toseban) [inline modules/ravestate_phrases_basic_en/__init__.py]: I love this - death to string referencing :crossed_swords:

File: modules/ravestate_phrases_basic_en/en/fillers.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 - "Well"
-- "So"
\ No newline at end of file
+- "Oh well"
+- "Interesting"
+- "Let me think"
+- "So"
+- "On that point"
+- "Uhm"
+- "Give me a moment"
```
Comment (Toseban) [inline modules/ravestate_phrases_basic_en/en/fillers.yml]: Hm, these are probably, mm, better

File: modules/ravestate_roboyio/__init__.py [hunk 3]
```diff
@@ -28,19 +28,30 @@
...
+
+        # @rs.state(cond=startup(), read=interloc.prop_all)
+        # def roboy_input(ctx: ContextWrapper):
+        #     while not ctx.shutting_down():
+        #         result = listen()
+        #         if result:
+        #             interloc.handle_single_interlocutor_input(ctx, result)
 
-        @state(read="rawio:out")
+        @rs.state(read="rawio:out")
```
Comment (Toseban) [inline modules/ravestate_roboyio/__init__.py]: Should be changed to rawio.prop_out

File: modules/ravestate_roboyio/__init__.py [hunk 4]
```diff
@@ -28,19 +28,30 @@
...
         def roboy_output(ctx):
-            ret = say(ctx["rawio:out:changed"])
-            logger.info(str(ret))
+            ret = say(unidecode(ctx["rawio:out:changed"]))
```
Comment (Toseban) [inline modules/ravestate_roboyio/__init__.py]: Should be changed to rawio.prop_out.changed()

File: modules/ravestate_ros2/ros2_properties.py [hunk 5]
```diff
@@ -22,28 +18,45 @@
...
 
 global_prop_set = set()
+global_node = None
+
+
+def set_node_once(ros2_node):
+    """
+    May be called once to override the internal ros2 node, which will
+     be created otherwise. ravestate_ros2 only supports one internal
+     node fro compatibility with ROS1<->ROS2 bridge.
```
Comment (Toseban) [inline modules/ravestate_ros2/ros2_properties.py]:
```suggestion
     node for compatibility with ROS1<->ROS2 bridge.
```

File: modules/ravestate_telegramio/telegram_bot.py [hunk 6]
```diff
@@ -58,32 +54,32 @@ def age(self):
...
 
-    @receptor(ctx_wrap=ctx, write="interloc:all")
-    def push_telegram_interloc(ctx: ContextWrapper, telegram_node: Node, name: str):
+    @rs.receptor(ctx_wrap=ctx, write=interloc.prop_all)
+    def push_telegram_interloc(ctx: rs.ContextWrapper, telegram_node: Node, name: str):
         """
         Push the telegram_node into interloc:all:name
         """
-        if ctx.push(parentpath="interloc:all", child=PropertyBase(name=name, default_value=telegram_node)):
+        if ctx.push(parent_property_or_path="interloc:all", child=rs.Property(name=name, default_value=telegram_node)):
```
Comment (Toseban) [inline modules/ravestate_telegramio/telegram_bot.py]:
```suggestion
        if ctx.push(parent_property_or_path=interloc.prop_all, child=rs.Property(name=name, default_value=telegram_node)):
```

File: config/finals_march_19.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+---
+module: idle
+config:
+  impatience_threshold: 10.0
+  bored_threshold: 1.0
+
+---
+module: telegramio
+config:
+  all_in_one_context: True
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply config/finals_march_19.yml]: Indeed not

File: modules/ravestate/__main__.py [hunk 8]
```diff
@@ -1,5 +1,8 @@
 from ravestate.context import Context
 import sys
 
+#from hanging_threads import start_monitoring
+#monitoring_thread = start_monitoring()
+
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply modules/ravestate/__main__.py]: Yes, I would like to keep it

File: modules/ravestate/activation.py [hunk 9]
```diff
@@ -270,40 +305,51 @@ def run(self, *args, **kwargs):
         Thread(target=self._run_private).start()
 
     def _reset_death_clock(self):
-        self.death_clock = self.ctx.lowest_upper_bound_eta(set(
-            sig for sig in self.constraint.signals() if not sig.spike))
+        # TODO: Proper impl. w/ user-defined Signal.wait(time)/waitTime()/Constraint.minWaitTime()
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply modules/ravestate/activation.py]: Will do

File: modules/ravestate/property.py [hunk 10]
```diff
@@ -98,10 +87,15 @@ def set_parent_path(self, path):
         """
         if not self.parent_path:
             self.parent_path = path
+            self.changed_signal.module_name = path
+            self.pushed_signal.module_name = path
+            self.popped_signal.module_name = path
+            self.true_signal.module_name = path
+            self.false_signal.module_name = path
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply modules/ravestate/property.py]: Yeah, this is definitely not just "module name" here... will change.

File: modules/ravestate_phrases_basic_en/__init__.py (hunk 1 above)
Comment (josephbirkner) [reply modules/ravestate_phrases_basic_en/__init__.py]: 🤜

File: modules/ravestate_phrases_basic_en/en/fillers.yml (hunk 2 above)
Comment (josephbirkner) [reply modules/ravestate_phrases_basic_en/en/fillers.yml]: On that point... 😆

File: modules/ravestate_roboyio/__init__.py (hunk 3 above)
Comment (josephbirkner) [reply modules/ravestate_roboyio/__init__.py]: Yes

File: modules/ravestate_roboyio/__init__.py (hunk 4 above)
Comment (josephbirkner) [reply modules/ravestate_roboyio/__init__.py]: Yes

**Pull request merged**